cmd/dockerd/docker.go

```diff
@@ -98,6 +100,7 @@ func main() {
 	if err != nil {
 		onError(err)
 	}
+	cmd.SetContext(ctx)
```

Comment:
My thinking here was to have an explicit context to set for the cycle of the process, instead of an implicit context created by the framework / library we're using. This would also allow for more explicit customisation of the context if there would be a need (which could happen in `main`).

(Hope my train of thought makes some sense, but happy to hear 😅)

Reply:
Explicit customization of context can be done with `cmd.ExecuteContext()`. I would prefer that over `cmd.SetContext` as it is more explicit about the scope of the operation bounded by the context: until the `ExecuteContext()` call returns.

Reply:
Ah, right, I changed that to `cmd.ExecuteContext()` in my last push. I wondered if you had an issue with constructing the context in main, but it if's only the `SetContext()` that should be fixed now 😅.